compiler/src/iree/compiler/Codegen/Common/GPU/GPUNestedLayoutDistributionPatterns.cpp

```diff
@@ -563,6 +563,223 @@ struct DistributeMultiReduction final
   int64_t maxBitsPerShuffle;
 };
 
+/// The lowering for Contract is performed in three steps (similar to above
+/// multi_reduction):
+///   1. Local Contract: Each thread performs operations on its locally
+///   distributed elements.
+///   2. Thread Reduction: Threads collectively reduce the results from step 1
+///   across threads,
+///      using a butterfly shuffle if distribution occurs along the reduction
+///      dimension.
+///   3. Accumulator Reduction: Each thread combines its intermediate results
+///   with its held accumulator.
+///
+/// Currently, reduction across multiple warps is not supported.
+/// TODO: Add support for reductions across multiple warps.
```

Comment:
Can you describe why we'd need it?

Reply:
Now the implementation is limited to support only 1 subgroup, I don't know whether 1 subgroup for sure is always the optimal config for performance.

Reply:
I think I added this comment for multi_reduction and it was copied over. Since we can describe reduction on subgroups on layouts, if we ever need workgroup reduce with atomic read/write we can do it. Can also just remove the comment if needed.

Reply:
Yes, I can remove it.